libV2/process.js

```diff
@@ -0,0 +1,38 @@
+const { EventEmitter } = require('events');
+const os = require('os');
+const { logger } = require('./utils');
+
+
+class Process extends EventEmitter {
+    async setup() {
+        const cleanUpFunc = this.join.bind(this);
+        ['SIGINT', 'SIGHUP', 'SIGQUIT', 'SIGTERM'].forEach(eventName => {
```

Comment:
not sure `SIGHUP` is a good candidate for termination, it's usually when the terminal attached to the process is closed, which would not make much sense if the process is a daemon.